### ISSUE TITLE
dataset issue

### ISSUE THREAD

**Issue by gejutaixiao22** [sunshangquan/Histoformer#13]
Hello, I would like to ask a question. When I was training the Outdoor-Rain dataset, the training set was 9,000 images and the test set was 600 images. When I ran the training, it showed a mismatch. Is there something wrong with my running process?
![图片6](https://github.com/user-attachments/assets/ad5e6990-4d00-4542-a3df-4c2679ca4eff)

**Comment by sunshangquan** [comment]
Hi @gejutaixiao22 , could you check the number of images at ```./Datasets/Outdoor-Rain/train/gt```? It seems that the error reports the number of training GT images is only 600, instead of 9,000.

**Comment by gejutaixiao22** [comment]
> 你好@gejutaixiao22，你能检查一下 处的图像数量`./Datasets/Outdoor-Rain/train/gt`吗？ 错误报告似乎训练 GT 图像的数量只有 600 张，而不是 9,000 张。

I am very sorry to bother you. I downloaded the Outdoor-Rain dataset, which has 9,000 images in the subfolder in and 600 images in the subfolder gt. I found that the images in the subfolder gt are different, while the images in the subfolder in have the same content every 15 images, except that the degree of degradation is different. Does this mean anything? I am very sorry to bother you.

**Comment by sunshangquan** [comment]
Oh I see. The 9,000 degraded images of Outdoor-Rain are synthesized on 600 clean images with 15 different degrees of degradation for each of them. You could try the following terminal script to duplicate GT images via soft-link and align the numbers of ```in``` and ```gt```:

```
cd ./Datasets/Outdoor-Rain/train
cur_path=`pwd`
mkdir gt_
for i in `ls in`; do ln -s $cur_path/gt/${i:0:7}".png" $cur_path/gt_/$i; done
``` 

Then use ```gt_```, instead of ```gt```, as your GT path in the configuration of the ```yml``` file.

**Comment by gejutaixiao22** [comment]
> 哦，我明白了。9,000张Outdoor-Rain的降级图像是在600张干净的图像上合成的，每张图像有15种不同的降级。您尝试以下脚本通过软链接可以复制GT图像并对齐`in`和数量`gt`：
> 
> ```
> cd ./Datasets/Outdoor-Rain/train
> cur_path=`pwd`
> mkdir gt_
> for i in `ls in`; do ln -s $cur_path/gt/${i:0:7}".png" $cur_path/gt_/$i; done
> ```
> 
> 然后在文件的配置中使用`gt_`，而不是作为您的 GT 路径。`gt``yml`

I tried running this script, thank you for your answer

**Comment by gejutaixiao22** [comment]
您好，我尝试运行了这个脚本。为什么会出现如下报错呢，可以解答一下么
![图片2](https://github.com/user-attachments/assets/2b410f63-3a62-45bf-94db-2ea3089986c7)

**Comment by sunshangquan** [comment]
可以尝试依次排查：1. 查看一下```./Datasets/Outdoor-Rain/train```底下是否有```gt_```路径；2. 把相对路径替换成绝对路径，也就是不要用```./```开头的路径，而是用```/home/```开头的路径，例如```/home/.../Datasets/Outdoor-Rain/train```。

**Comment by gejutaixiao22** [comment]
谢谢你的解惑，已经解决了

**Comment by gejutaixiao22** [comment]
你好，我在跑RainDrop的时候也遇到了类似的问题。我发现data文件夹里和gt文件夹里的图片命名是不一样的，导致了以下的错误，请问这种要怎么解决。
![图片3](https://github.com/user-attachments/assets/886e0511-5598-44b2-96c4-49e1d15e3dcb)

**Comment by gejutaixiao22** [comment]
我把图片名字更改了，问题已经解决

**Comment by sunshangquan** [comment]
Glad for you:)
